### PR TITLE
Added priority queue design pattern 

### PR DESCRIPTION
**Pull request by ranjeet-floyd**
Added priority queue design pattern

- Application posting a message can assign a priority to a message and the messages in the queue are automatically reordered so that messages with a higher priority will be received before those of a lower priority.
- Issue - #449

Pull request description

- Implemented https://docs.microsoft.com/en-us/previous-versions/msp-n-p/dn589794(v=pandp.10)

### CODE REVIEW

**Comment by iluwatar** [comment]
There's a conflict in `pom.xml` that prevents merging this PR

File: priority-queue/src/main/java/com/iluwatar/priority/queue/Message.java [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+ * THE SOFTWARE.
+ */
+package com.iluwatar.priority.queue;
+
+/**
+ * Simple Bean
+ */
+public class Message implements Comparable<Message> {
+  private String message;
+  private int priority; // define message priority in queue
```
Comment (iluwatar) [inline priority-queue/src/main/java/com/iluwatar/priority/queue/Message.java]: Make these final

File: priority-queue/pom.xml [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+    <build>
+        <plugins>
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-compiler-plugin</artifactId>
+                <configuration>
+                    <source>8</source>
+                    <target>8</target>
+                </configuration>
+            </plugin>
```
Comment (iluwatar) [inline priority-queue/pom.xml]: Why is maven-compiler-plugin needed here?

File: priority-queue/src/main/java/com/iluwatar/priority/queue/Application.java [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package com.iluwatar.priority.queue;
+
+/**
+ * Example
+ */
```
Comment (iluwatar) [inline priority-queue/src/main/java/com/iluwatar/priority/queue/Application.java]: Explain here more verbosely how the example works and implements the pattern

File: priority-queue/src/main/java/com/iluwatar/priority/queue/Message.java (hunk 1 above)
Comment (ranjeet-floyd) [reply priority-queue/src/main/java/com/iluwatar/priority/queue/Message.java]: Ok

File: priority-queue/pom.xml (hunk 2 above)
Comment (ranjeet-floyd) [reply priority-queue/pom.xml]: I will check. Thanks

File: priority-queue/src/main/java/com/iluwatar/priority/queue/Application.java (hunk 3 above)
Comment (ranjeet-floyd) [reply priority-queue/src/main/java/com/iluwatar/priority/queue/Application.java]: i will add better comment. Thanks

File: priority-queue/pom.xml (hunk 2 above)
Comment (ranjeet-floyd) [reply priority-queue/pom.xml]: Removed

File: priority-queue/src/main/java/com/iluwatar/priority/queue/Application.java (hunk 3 above)
Comment (ranjeet-floyd) [reply priority-queue/src/main/java/com/iluwatar/priority/queue/Application.java]: Added better message

File: priority-queue/src/main/java/com/iluwatar/priority/queue/Message.java (hunk 1 above)
Comment (ranjeet-floyd) [reply priority-queue/src/main/java/com/iluwatar/priority/queue/Message.java]: Done

**Comment by ranjeet-floyd** [comment]
> There's a conflict in `pom.xml` that prevents merging this PR

Fixed

**Comment by iluwatar** [comment]
Good job @ranjeet-floyd :+1:

**Pull request merged**